core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala

```diff
@@ -590,6 +627,48 @@ class TiRegionPartitioner(regions: List[TiRegion]) extends Partitioner {
   }
 }
 
+class keyWithDupInfo(var key: SerializableKey, var oldRow: TiRow) extends Serializable {
+  var oldHandle: Long = -1L
```

Comment:
add `private`?

Reply:
why not add `oldHandle` to Constructor function?

Reply:
It is only used to detect a conflict.

Reply:
if it is possible please
1. add `oldHandle` to Constructor function
2. change `var` to `val`

like
`class keyWithDupInfo(val key: SerializableKey, val oldRow: TiRow, val oldHandle: Long)`